diff --git a/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj b/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
index 77ce1dab..f6a20dfa 100644
--- a/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
+++ b/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
@@ -139,13 +139,13 @@
 		69157C432471A5A1005B9002 /* items_es.json in Resources */ = {isa = PBXBuildFile; fileRef = 699425E5246927C40089C21B /* items_es.json */; };
 		69157C532471A5A1005B9002 /* tools in Resources */ = {isa = PBXBuildFile; fileRef = 69CFCA86245AAF520059C067 /* tools */; };
 		69157C5C2471A5C1005B9002 /* CloudKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 69157C5B2471A5C1005B9002 /* CloudKit.framework */; };
-		693352842473C4B5007EF9AB /* TodaySpecialCharactersSection.swift in Sources */ = {isa = PBXBuildFile; fileRef = 693352832473C4B5007EF9AB /* TodaySpecialCharactersSection.swift */; };
 		696B454824791EDE00E16252 /* islanddevelopment in Resources */ = {isa = PBXBuildFile; fileRef = 696B454724791EDE00E16252 /* islanddevelopment */; };
 		696B454A247A633D00E16252 /* TodayMusicPlayerSection.swift in Sources */ = {isa = PBXBuildFile; fileRef = 696B4549247A633D00E16252 /* TodayMusicPlayerSection.swift */; };
 		696B454C247A948A00E16252 /* MusicPlayerManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = 696B454B247A948A00E16252 /* MusicPlayerManager.swift */; };
 		69867DF1247C136900C6A759 /* CollectionProgressRowViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69867DF0247C136900C6A759 /* CollectionProgressRowViewModel.swift */; };
 		69CCD28A247D9FD000DF381B /* ItemBellsView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69CCD289247D9FD000DF381B /* ItemBellsView.swift */; };
 		69CCD28C247DA4D100DF381B /* ItemStyleBadgeView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69CCD28B247DA4D100DF381B /* ItemStyleBadgeView.swift */; };
+		69CCD293247ECDD000DF381B /* TodaySpecialCharactersSection.swift in Sources */ = {isa = PBXBuildFile; fileRef = 693352832473C4B5007EF9AB /* TodaySpecialCharactersSection.swift */; };
 		69CFCAC9245ADC780059C067 /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69CFCAC8245ADC780059C067 /* AppDelegate.swift */; };
 		69CFCACD245ADC7A0059C067 /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 69CFCACC245ADC7A0059C067 /* Assets.xcassets */; };
 		69CFCAD0245ADC7A0059C067 /* Preview Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 69CFCACF245ADC7A0059C067 /* Preview Assets.xcassets */; };
@@ -172,6 +172,15 @@
 		69E4BBB6245AE5C9001035E2 /* ItemDetailView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69E4BBB5245AE5C9001035E2 /* ItemDetailView.swift */; };
 		69E4BBB7245AE60C001035E2 /* Color.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69AF8C7B243E4D29001DD9CB /* Color.swift */; };
 		69E4BBB8245AE60F001035E2 /* Colors.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 69AF8C79243E4C78001DD9CB /* Colors.xcassets */; };
+		AE0B5B5A247AD6DF0075CF16 /* DesignFormView.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE0B5B59247AD6DF0075CF16 /* DesignFormView.swift */; };
+		AE0B5B5C247AD7590075CF16 /* DesignFormViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE0B5B5B247AD7590075CF16 /* DesignFormViewModel.swift */; };
+		AE0B5B5E247B14120075CF16 /* DesignRowViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE0B5B5D247B14120075CF16 /* DesignRowViewModel.swift */; };
+		AE2D7EF8247E8C9A006C1F1A /* CollectionMoreDetailView.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE2D7EF7247E8C9A006C1F1A /* CollectionMoreDetailView.swift */; };
+		AE2D7EFA247E8CBD006C1F1A /* CollectionMoreDetailViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE2D7EF9247E8CBD006C1F1A /* CollectionMoreDetailViewModel.swift */; };
+		AE2D7EFC247E9726006C1F1A /* DesignListView.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE2D7EFB247E9726006C1F1A /* DesignListView.swift */; };
+		AE2D7EFE247EAF05006C1F1A /* CrittersListView.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE2D7EFD247EAF05006C1F1A /* CrittersListView.swift */; };
+		AE2D7F01247EB2A8006C1F1A /* MessageView.swift in Sources */ = {isa = PBXBuildFile; fileRef = AE2D7F00247EB2A8006C1F1A /* MessageView.swift */; };
+		AEEEC4C6247AB1F400894C02 /* DesignRowView.swift in Sources */ = {isa = PBXBuildFile; fileRef = AEEEC4C5247AB1F400894C02 /* DesignRowView.swift */; };
 		FA0EC271247B77E400447D7E /* VillagersSortView.swift in Sources */ = {isa = PBXBuildFile; fileRef = FA0EC270247B77E400447D7E /* VillagersSortView.swift */; };
 /* End PBXBuildFile section */
 
@@ -344,6 +353,15 @@
 		69E4BBBA245B0BEF001035E2 /* AboutView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = AboutView.swift; sourceTree = "<group>"; };
 		69E98C6E247439AD00A200FE /* it */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = it; path = it.lproj/Localizable.strings; sourceTree = "<group>"; };
 		69EC82ED2461E79D00CE9064 /* TurnipsChartTopLegendView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = TurnipsChartTopLegendView.swift; sourceTree = "<group>"; };
+		AE0B5B59247AD6DF0075CF16 /* DesignFormView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DesignFormView.swift; sourceTree = "<group>"; };
+		AE0B5B5B247AD7590075CF16 /* DesignFormViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DesignFormViewModel.swift; sourceTree = "<group>"; };
+		AE0B5B5D247B14120075CF16 /* DesignRowViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DesignRowViewModel.swift; sourceTree = "<group>"; };
+		AE2D7EF7247E8C9A006C1F1A /* CollectionMoreDetailView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CollectionMoreDetailView.swift; sourceTree = "<group>"; };
+		AE2D7EF9247E8CBD006C1F1A /* CollectionMoreDetailViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CollectionMoreDetailViewModel.swift; sourceTree = "<group>"; };
+		AE2D7EFB247E9726006C1F1A /* DesignListView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DesignListView.swift; sourceTree = "<group>"; };
+		AE2D7EFD247EAF05006C1F1A /* CrittersListView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CrittersListView.swift; sourceTree = "<group>"; };
+		AE2D7F00247EB2A8006C1F1A /* MessageView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = MessageView.swift; sourceTree = "<group>"; };
+		AEEEC4C5247AB1F400894C02 /* DesignRowView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DesignRowView.swift; sourceTree = "<group>"; };
 		FA0EC270247B77E400447D7E /* VillagersSortView.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = VillagersSortView.swift; sourceTree = "<group>"; };
 /* End PBXFileReference section */
 
@@ -457,6 +475,7 @@
 			isa = PBXGroup;
 			children = (
 				693E0B21244C596D00D20E46 /* ActiveCrittersView.swift */,
+				AE2D7EFD247EAF05006C1F1A /* CrittersListView.swift */,
 			);
 			path = critters;
 			sourceTree = "<group>";
@@ -628,21 +647,23 @@
 		69AF8C64243E2857001DD9CB /* views */ = {
 			isa = PBXGroup;
 			children = (
-				69157B6F2471226B005B9002 /* mysteryIslands */,
-				024363CF2465E7FD006A37A0 /* todayDashboard */,
-				69875F5324654EA400B1D46C /* userLists */,
-				693E7292245D833F00CD26F4 /* subscription */,
 				69E4BBB9245B0BE4001035E2 /* about */,
-				693E0B20244C596000D20E46 /* critters */,
-				697E63FE244AD664008FB710 /* settings */,
-				687B5765244A02E400C1E86F /* turnips */,
-				697E63F32449A488008FB710 /* villagers */,
-				6942FC38244989470062FE8F /* listing */,
 				69C5F5E8244217EF00BF8FA4 /* categories */,
 				69BEA983243E638C00F02C61 /* collection */,
+				693E0B20244C596000D20E46 /* critters */,
+				AEEEC4C4247AB1B100894C02 /* design */,
+				69BEA980243E608900F02C61 /* home */,
 				69BEA982243E609400F02C61 /* items */,
+				6942FC38244989470062FE8F /* listing */,
+				AE2D7EFF247EB289006C1F1A /* message */,
+				69157B6F2471226B005B9002 /* mysteryIslands */,
+				697E63FE244AD664008FB710 /* settings */,
 				69BEA981243E608E00F02C61 /* shared */,
-				69BEA980243E608900F02C61 /* home */,
+				693E7292245D833F00CD26F4 /* subscription */,
+				024363CF2465E7FD006A37A0 /* todayDashboard */,
+				687B5765244A02E400C1E86F /* turnips */,
+				69875F5324654EA400B1D46C /* userLists */,
+				697E63F32449A488008FB710 /* villagers */,
 			);
 			path = views;
 			sourceTree = "<group>";
@@ -650,16 +671,19 @@
 		69AF8C6B243E2A0C001DD9CB /* viewModels */ = {
 			isa = PBXGroup;
 			children = (
-				69875F5A24655AF000B1D46C /* UserListDetailViewModel.swift */,
-				69875F5624654F7100B1D46C /* UserListFormViewModel.swift */,
-				68BA5829244B693300B53811 /* TurnipsViewModel.swift */,
+				6941F3872452BDDD00EC2E02 /* CategoriesSearchViewModel.swift */,
+				AE2D7EF9247E8CBD006C1F1A /* CollectionMoreDetailViewModel.swift */,
+				69867DF0247C136900C6A759 /* CollectionProgressRowViewModel.swift */,
 				693E0B1C244C315B00D20E46 /* DashboardViewModel.swift */,
-				69AF8C6C243E2A18001DD9CB /* ItemsViewModel.swift */,
+				AE0B5B5B247AD7590075CF16 /* DesignFormViewModel.swift */,
+				AE0B5B5D247B14120075CF16 /* DesignRowViewModel.swift */,
 				68DFFFAC24492D7F00DEA07F /* ItemDetailViewModel.swift */,
-				697E63F62449A7BA008FB710 /* VillagersViewModel.swift */,
-				6941F3872452BDDD00EC2E02 /* CategoriesSearchViewModel.swift */,
+				69AF8C6C243E2A18001DD9CB /* ItemsViewModel.swift */,
+				68BA5829244B693300B53811 /* TurnipsViewModel.swift */,
+				69875F5A24655AF000B1D46C /* UserListDetailViewModel.swift */,
+				69875F5624654F7100B1D46C /* UserListFormViewModel.swift */,
 				695D45FA24671B1800B5C313 /* VillagerDetailViewModel.swift */,
-				69867DF0247C136900C6A759 /* CollectionProgressRowViewModel.swift */,
+				697E63F62449A7BA008FB710 /* VillagersViewModel.swift */,
 			);
 			path = viewModels;
 			sourceTree = "<group>";
@@ -724,6 +748,7 @@
 			isa = PBXGroup;
 			children = (
 				69BEA984243E639900F02C61 /* CollectionListView.swift */,
+				AE2D7EF7247E8C9A006C1F1A /* CollectionMoreDetailView.swift */,
 				6903CD0E246C7E0300150FA1 /* CollectionProgressDetailView.swift */,
 				6903CD10246C7EB800150FA1 /* CollectionProgressRow.swift */,
 				4F92A4522470AC01003A2911 /* CollectionRowView.swift */,
@@ -835,6 +860,24 @@
 			path = about;
 			sourceTree = "<group>";
 		};
+		AE2D7EFF247EB289006C1F1A /* message */ = {
+			isa = PBXGroup;
+			children = (
+				AE2D7F00247EB2A8006C1F1A /* MessageView.swift */,
+			);
+			path = message;
+			sourceTree = "<group>";
+		};
+		AEEEC4C4247AB1B100894C02 /* design */ = {
+			isa = PBXGroup;
+			children = (
+				AE0B5B59247AD6DF0075CF16 /* DesignFormView.swift */,
+				AEEEC4C5247AB1F400894C02 /* DesignRowView.swift */,
+				AE2D7EFB247E9726006C1F1A /* DesignListView.swift */,
+			);
+			path = design;
+			sourceTree = "<group>";
+		};
 /* End PBXGroup section */
 
 /* Begin PBXNativeTarget section */
@@ -1003,7 +1046,9 @@
 			isa = PBXSourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				69CCD293247ECDD000DF381B /* TodaySpecialCharactersSection.swift in Sources */,
 				69157B7E2471A5A1005B9002 /* TodayMysteryIslandsSection.swift in Sources */,
+				AEEEC4C6247AB1F400894C02 /* DesignRowView.swift in Sources */,
 				69157B7F2471A5A1005B9002 /* TodayNookazonSection.swift in Sources */,
 				69157B802471A5A1005B9002 /* UserListFormView.swift in Sources */,
 				69157B812471A5A1005B9002 /* RowLoadingView.swift in Sources */,
@@ -1015,11 +1060,13 @@
 				69157B862471A5A1005B9002 /* MysteryIslandRow.swift in Sources */,
 				69157B872471A5A1005B9002 /* TurnipsChartAverageCurve.swift in Sources */,
 				69157B882471A5A1005B9002 /* CategoryDetailView.swift in Sources */,
+				AE2D7EFA247E8CBD006C1F1A /* CollectionMoreDetailViewModel.swift in Sources */,
 				69157B892471A5A1005B9002 /* SettingsView.swift in Sources */,
 				69157B8A2471A5A1005B9002 /* IslandDetailViewModel.swift in Sources */,
 				69157B8B2471A5A1005B9002 /* DocumentPickerView.swift in Sources */,
 				69157B8C2471A5A1005B9002 /* TodayTurnipSection.swift in Sources */,
 				69157B8D2471A5A1005B9002 /* UserListDetailViewModel.swift in Sources */,
+				AE0B5B5E247B14120075CF16 /* DesignRowViewModel.swift in Sources */,
 				69157B8E2471A5A1005B9002 /* SectionHeaderView.swift in Sources */,
 				69157B8F2471A5A1005B9002 /* IslandDetailView.swift in Sources */,
 				69157B902471A5A1005B9002 /* UserListDetailView.swift in Sources */,
@@ -1043,12 +1090,14 @@
 				4C16FB41247AC9B0009F24E3 /* GridStack.swift in Sources */,
 				69157B9F2471A5A1005B9002 /* SafariView.swift in Sources */,
 				69157BA02471A5A1005B9002 /* DismissingKeyboardOnSwipe.swift in Sources */,
+				AE0B5B5A247AD6DF0075CF16 /* DesignFormView.swift in Sources */,
 				69157BA12471A5A1005B9002 /* ItemDetailSeasonSectionView.swift in Sources */,
 				69157BA22471A5A1005B9002 /* MysteryIslandsList.swift in Sources */,
 				69157BA32471A5A1005B9002 /* TurnipsChartValuesView.swift in Sources */,
 				69157BA42471A5A1005B9002 /* VillagerRowView.swift in Sources */,
 				69CCD28A247D9FD000DF381B /* ItemBellsView.swift in Sources */,
 				69157BA52471A5A1005B9002 /* AdaptsToSoftwareKeyboard.swift in Sources */,
+				AE2D7EF8247E8C9A006C1F1A /* CollectionMoreDetailView.swift in Sources */,
 				69157BA62471A5A1005B9002 /* CollectionProgressRow.swift in Sources */,
 				69157BA72471A5A1005B9002 /* TurnipIslandRow.swift in Sources */,
 				69157BA82471A5A1005B9002 /* CategoryHeaderView.swift in Sources */,
@@ -1077,17 +1126,19 @@
 				69157BBE2471A5A1005B9002 /* ItemsListView.swift in Sources */,
 				69157BBF2471A5A1005B9002 /* ItemDetailView.swift in Sources */,
 				69157BC02471A5A1005B9002 /* TodayBirthdaysSection.swift in Sources */,
+				AE2D7EFC247E9726006C1F1A /* DesignListView.swift in Sources */,
 				69157BC12471A5A1005B9002 /* ItemsViewModel.swift in Sources */,
 				69157BC22471A5A1005B9002 /* TodayEventsSection.swift in Sources */,
 				69157BC32471A5A1005B9002 /* FeedbackGenerator.swift in Sources */,
 				69157BC42471A5A1005B9002 /* CategoryRowView.swift in Sources */,
+				AE2D7EFE247EAF05006C1F1A /* CrittersListView.swift in Sources */,
 				69157BC52471A5A1005B9002 /* TodayCollectionProgressSection.swift in Sources */,
 				69157BC62471A5A1005B9002 /* SubscribeView.swift in Sources */,
 				69157BC72471A5A1005B9002 /* TurnipsFormView.swift in Sources */,
 				69157BC82471A5A1005B9002 /* TurnipsChartMinMaxCurves.swift in Sources */,
 				69157BC92471A5A1005B9002 /* TurnipsChartGrid.swift in Sources */,
 				69157BCA2471A5A1005B9002 /* CollectionRowView.swift in Sources */,
-				693352842473C4B5007EF9AB /* TodaySpecialCharactersSection.swift in Sources */,
+				AE0B5B5C247AD7590075CF16 /* DesignFormViewModel.swift in Sources */,
 				69157BCB2471A5A1005B9002 /* SearchField.swift in Sources */,
 				69157BCC2471A5A1005B9002 /* CollectionProgressDetailView.swift in Sources */,
 				69157BCD2471A5A1005B9002 /* AppIconPickerView.swift in Sources */,
@@ -1102,6 +1153,7 @@
 				69157BD72471A5A1005B9002 /* TodaySectionEditView.swift in Sources */,
 				69157BD82471A5A1005B9002 /* VillagersViewModel.swift in Sources */,
 				FA0EC271247B77E400447D7E /* VillagersSortView.swift in Sources */,
+				AE2D7F01247EB2A8006C1F1A /* MessageView.swift in Sources */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 		};
diff --git a/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings
index c2835050..fde20d49 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings
@@ -183,6 +183,7 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Items" = "Objets";
 "Critters" = "Créatures";
 "Lists" = "Listes";
+"More" = "Plus";
 
 // ItemsListView
 "Clear Selection" = "Enlever la séléction";
diff --git a/ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/environments/UserCollection.swift b/ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/environments/UserCollection.swift
index 3ecc96d7..145439af 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/environments/UserCollection.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/environments/UserCollection.swift
@@ -19,6 +19,7 @@ public class UserCollection: ObservableObject {
     @Published public var villagers: [Villager] = []
     @Published public var critters: [Item] = []
     @Published public var lists: [UserList] = []
+    @Published public var designs: [Design] = []
     @Published public var dailyTasks = DailyTasks()
     
     @Published public var isCloudEnabled = true
@@ -32,6 +33,7 @@ public class UserCollection: ObservableObject {
         let critters: [Item]
         let lists: [UserList]?
         let dailyTasks: DailyTasks?
+        let designs: [Design]?
     }
     
     private let filePath: URL
@@ -176,7 +178,19 @@ public class UserCollection: ObservableObject {
         lists[index].items.remove(at: at)
         save()
     }
-    
+
+    // MARK: - Custom Designs
+
+    public func addDesign(_ design: Design) {
+        designs.append(design)
+        save()
+    }
+
+    public func deleteDesign(at design: Int) {
+        designs.remove(at: design)
+        save()
+    }
+
     // MARK: - CloudKit
     private func checkiCloudStatus() {
         CKContainer.default().accountStatus { (status, error) in
@@ -267,7 +281,8 @@ public class UserCollection: ObservableObject {
                                           villagers: weakself.villagers,
                                           critters: weakself.critters,
                                           lists: weakself.lists,
-                                          dailyTasks: weakself.dailyTasks)
+                                          dailyTasks: weakself.dailyTasks,
+                                          designs: weakself.designs)
                 let data = try weakself.encoder.encode(savedData)
                 try data.write(to: weakself.filePath, options: .atomicWrite)
                 
@@ -295,6 +310,7 @@ public class UserCollection: ObservableObject {
                 self.critters = savedData.critters
                 self.lists = savedData.lists ?? []
                 self.dailyTasks = savedData.dailyTasks ?? DailyTasks()
+                self.designs = savedData.designs ?? []
                 return true
             } catch {
                 return false
@@ -311,6 +327,7 @@ public class UserCollection: ObservableObject {
             self.critters = []
             self.lists = []
             self.dailyTasks = DailyTasks()
+            self.designs = []
             save()
             return true
         } catch {
diff --git a/ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/models/Design.swift b/ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/models/Design.swift
new file mode 100644
index 00000000..e33a4f06
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/packages/Backend/Sources/Backend/models/Design.swift
@@ -0,0 +1,48 @@
+//
+//  Design.swift
+//  
+//  Created by Otavio Cordeiro on 24.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import Foundation
+
+public struct Design: Identifiable, Codable, Equatable {
+
+    // MARK: - Properties
+
+    public let id = UUID()
+    public var title: String = ""
+    public var code: String = ""
+    public var description: String = ""
+
+    public var hasValidCode: Bool {
+        let validPrefixes: Set<String> = ["MA", "MO"]
+
+        let simplifiedCode = code
+            .uppercased()
+            .unicodeScalars
+            .filter(CharacterSet.alphanumerics.contains)
+
+        guard
+            simplifiedCode.count == 14,
+            case let prefix = String(simplifiedCode.prefix(2)),
+            validPrefixes.contains(prefix) else { return false }
+
+        return true
+    }
+
+    // MARK: - Life cycle
+
+    public init(title: String = "", code: String = "", description: String = "") {
+        self.title = title
+        self.code = code
+        self.description = description
+    }
+
+    // MARK: - Equatable
+
+    public static func == (lhs: Design, rhs: Design) -> Bool {
+        lhs.id == rhs.id
+    }
+}
diff --git a/ACHNBrowserUI/ACHNBrowserUI/viewModels/CollectionMoreDetailViewModel.swift b/ACHNBrowserUI/ACHNBrowserUI/viewModels/CollectionMoreDetailViewModel.swift
new file mode 100644
index 00000000..ed7ceb0f
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/viewModels/CollectionMoreDetailViewModel.swift
@@ -0,0 +1,30 @@
+//
+//  MoreViewModel.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 27.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import Backend
+
+struct CollectionMoreDetailViewModel {
+
+    // MARK: - Types
+
+    enum Row: String, CaseIterable, CustomStringConvertible {
+        case critters
+        case designs
+
+        var description: String {
+            switch self {
+            case .critters: return "Critters"
+            case .designs: return "Creators and Designs"
+            }
+        }
+    }
+
+    // MARK: - Properties
+
+    let rows = Row.allCases
+}
diff --git a/ACHNBrowserUI/ACHNBrowserUI/viewModels/DesignFormViewModel.swift b/ACHNBrowserUI/ACHNBrowserUI/viewModels/DesignFormViewModel.swift
new file mode 100644
index 00000000..a358b190
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/viewModels/DesignFormViewModel.swift
@@ -0,0 +1,33 @@
+//
+//  DesignFormViewModel.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 24.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import Foundation
+import Backend
+
+class DesignFormViewModel: ObservableObject {
+
+    // MARK: - Properties
+
+    @Published var design: Design
+
+    // MARK: - Life cycle
+
+    init(design: Design?) {
+        if let design = design {
+            self.design = design
+        } else {
+            self.design = Design()
+        }
+    }
+
+    // MARK: - Public
+
+    func save() {
+        UserCollection.shared.addDesign(design)
+    }
+}
diff --git a/ACHNBrowserUI/ACHNBrowserUI/viewModels/DesignRowViewModel.swift b/ACHNBrowserUI/ACHNBrowserUI/viewModels/DesignRowViewModel.swift
new file mode 100644
index 00000000..b7fa0692
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/viewModels/DesignRowViewModel.swift
@@ -0,0 +1,90 @@
+//
+//  DesignRowViewModel.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 24.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import Foundation
+import Backend
+
+struct DesignRowViewModel {
+
+    // MARK: - Properties
+
+    private let design: Design
+
+    public var title: String {
+        design.title
+    }
+
+    public var description: String {
+        design.description
+    }
+
+    public var code: String {
+        formatted(code: design.code)
+    }
+
+    public var category: String {
+        category(for: design.code)
+    }
+
+    // MARK: - Life cycle
+
+    init(design: Design) {
+        self.design = design
+    }
+
+    // MARK: - Private
+
+    /// Formats the creator/item code to the format used by Nintendo.
+    ///
+    /// - Creator: MA-XXXX-XXXX-XXXX
+    /// - Item: MO-XXXX-XXXX-XXXX
+    ///
+    /// - Parameter code: The code to be formatted.
+    /// - Returns: Formatted string representing the given code.
+    private func formatted(code: String) -> String {
+        let codeFormat = "XX-XXXX-XXXX-XXXX"
+
+        let filteredUnicodeScalars = code
+            .uppercased()
+            .unicodeScalars
+            .filter(CharacterSet.alphanumerics.contains)
+
+        let simplifiedCode = String(filteredUnicodeScalars)
+
+        var result = ""
+        var index = simplifiedCode.startIndex
+        for character in codeFormat where index < simplifiedCode.endIndex {
+            if character == "X" {
+                result.append(simplifiedCode[index])
+                index = codeFormat.index(after: index)
+            } else {
+                result.append(character)
+            }
+        }
+
+        return result
+    }
+
+    /// Category based on code
+    ///
+    ///  - MA-XXXX-XXXX-XXXX -> Creator
+    ///  - MO-XXXX-XXXX-XXXX -> Item
+    ///
+    /// - Parameter code: The code to be categorized.
+    /// - Returns: The category based on the given code.
+    private func category(for code: String) -> String {
+        let prefix = String(code.uppercased().prefix(2))
+
+        switch prefix {
+        case "MA":
+            return "Creator"
+        default:
+            return "Item"
+        }
+    }
+}
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionListView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionListView.swift
index 8c7665b7..a6907cbc 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionListView.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionListView.swift
@@ -11,7 +11,7 @@ import SwiftUIKit
 import Backend
 
 enum Tabs: String, CaseIterable {
-    case items, villagers, critters, lists
+    case items, villagers, lists, more
 }
 
 struct CollectionListView: View {
@@ -19,11 +19,11 @@ struct CollectionListView: View {
     @EnvironmentObject private var subscriptionManager: SubscriptionManager
     @State private var selectedTab: Tabs = .items
     @State private var sheet: Sheet.SheetType?
-        
+
     private var categories: [String] {
         Array(Set(collection.items.map(\.category))).sorted()
     }
-    
+
     var body: some View {
         NavigationView {
             List {
@@ -38,15 +38,11 @@ struct CollectionListView: View {
                                 VillagerRowView(villager: villager)
                             }
                         }
-                    } else if selectedTab == .critters && !collection.critters.isEmpty {
-                        ForEach(collection.critters) { critter in
-                            NavigationLink(destination: ItemDetailView(item: critter)) {
-                                ItemRowView(displayMode: .large, item: critter)
-                            }
-                        }
                     } else if selectedTab == .lists {
                         userListsSections
-                    }  else {
+                    } else if selectedTab == .more {
+                        CollectionMoreDetailView(viewModel: CollectionMoreDetailViewModel())
+                    } else {
                         emptyView
                     }
                 }
@@ -63,7 +59,7 @@ struct CollectionListView: View {
             }
         }
     }
-    
+
     private var userListsSections: some View {
         Group {
             if subscriptionManager.subscriptionStatus == .subscribed || collection.lists.isEmpty {
@@ -85,18 +81,17 @@ struct CollectionListView: View {
             }
         }
     }
-    
+
     private var placeholderView: some View {
         Text("Please select or go stars some items!")
             .foregroundColor(.acSecondaryText)
     }
-    
+
     private var emptyView: some View {
         let selectedTabName = NSLocalizedString(selectedTab.rawValue, comment: "")
-        return Text("When you stars some \(selectedTabName), they'll be displayed here.")
-            .foregroundColor(.acSecondaryText)
+        return MessageView(collectionName: selectedTabName)
     }
-    
+
     private var picker: some View {
         Picker(selection: $selectedTab, label: Text("")) {
             ForEach(Tabs.allCases, id: \.self) { tab in
@@ -117,5 +112,3 @@ struct CollectionListView_Previews: PreviewProvider {
         .environmentObject(SubscriptionManager.shared)
     }
 }
-
-
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionMoreDetailView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionMoreDetailView.swift
new file mode 100644
index 00000000..ccc6d4e3
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/collection/CollectionMoreDetailView.swift
@@ -0,0 +1,62 @@
+//
+//  MoreView.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 27.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import SwiftUI
+import Backend
+
+struct CollectionMoreDetailView: View {
+
+    // MARK: - Properties
+
+    @EnvironmentObject private var collection: UserCollection
+
+    let viewModel: CollectionMoreDetailViewModel
+
+    // MARK: - Life cycle
+
+    init(viewModel: CollectionMoreDetailViewModel) {
+        self.viewModel = viewModel
+    }
+
+    // MARK: - Public
+
+    var body: some View {
+        ForEach(viewModel.rows, id: \.self) { row in
+            self.makeRowView(for: row)
+        }
+    }
+
+    // MARK: - Private
+
+    private func makeRowView(for row: CollectionMoreDetailViewModel.Row) -> some View {
+        switch row {
+        case .critters:
+            return NavigationLink(destination: CrittersListView()) {
+                Text(row.description)
+            }.eraseToAnyView()
+
+        case .designs:
+            return NavigationLink(destination: DesignListView()) {
+                Text(row.description)
+            }.eraseToAnyView()
+        }
+    }
+}
+
+// MARK: - Preview
+
+#if DEBUG
+struct MoreView_Previews: PreviewProvider {
+    static var previews: some View {
+        List {
+            CollectionMoreDetailView(viewModel: CollectionMoreDetailViewModel())
+        }
+        .previewLayout(.sizeThatFits)
+    }
+}
+#endif
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/critters/CrittersListView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/critters/CrittersListView.swift
new file mode 100644
index 00000000..1298d2a0
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/critters/CrittersListView.swift
@@ -0,0 +1,47 @@
+//
+//  CrittersListView.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 27.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import SwiftUI
+import Backend
+
+struct CrittersListView: View {
+
+    // MARK: - Properties
+
+    @EnvironmentObject private var collection: UserCollection
+
+    // MARK: - Public
+
+    var body: some View {
+        List {
+            if !collection.critters.isEmpty {
+                ForEach(collection.critters) { critter in
+                    NavigationLink(destination: ItemDetailView(item: critter)) {
+                        ItemRowView(displayMode: .large, item: critter)
+                    }
+                }
+            } else {
+                MessageView(collectionName: "critters")
+            }
+        }
+        .navigationBarTitle(Text("Critters"), displayMode: .automatic)
+    }
+}
+
+// MARK: - Preview
+
+#if DEBUG
+struct CrittersListView_Previews: PreviewProvider {
+    static var previews: some View {
+        NavigationView {
+            CrittersListView()
+        }
+        .environmentObject(UserCollection.shared)
+    }
+}
+#endif
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignFormView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignFormView.swift
new file mode 100644
index 00000000..8b3c876f
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignFormView.swift
@@ -0,0 +1,122 @@
+//
+//  DesignFormView.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 24.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import SwiftUI
+import SwiftUIKit
+import Backend
+
+struct DesignFormView: View {
+
+    // MARK: - Properties
+
+    @Environment(\.presentationMode) private var presentationMode
+    @ObservedObject private var viewModel: DesignFormViewModel
+    @State private var nameErrorBorderColor: Color = .clear
+    @State private var codeErrorBorderColor: Color = .clear
+
+    // MARK: - Life cycle
+
+    init(viewModel: DesignFormViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        NavigationView {
+            Form {
+                HStack {
+                    Text("Name")
+                    Spacer()
+                    TextField("Creator/Item name",
+                              text: $viewModel.design.title,
+                              onEditingChanged: { _ in self.nameErrorBorderColor = .clear })
+                        .foregroundColor(.acText)
+                }
+                .border(nameErrorBorderColor)
+
+                HStack {
+                    Text("Code")
+                    Spacer()
+                    TextField("MX-XXXX-XXXX-XXXX",
+                              text: $viewModel.design.code,
+                              onEditingChanged: { _ in self.codeErrorBorderColor = .clear })
+                        .foregroundColor(.acText)
+                }
+                .border(codeErrorBorderColor)
+
+                HStack {
+                    Text("Description")
+                    Spacer()
+                    TextField("Short description about creator/item",
+                              text: $viewModel.design.description)
+                        .foregroundColor(.acText)
+                }
+            }
+            .navigationBarTitle("Add design")
+            .navigationBarItems(leading: dismissButton, trailing: saveButton)
+        }
+        .navigationViewStyle(StackNavigationViewStyle())
+    }
+
+    // MARK: - Private
+
+    private var dismissButton: some View {
+        Button(action: {
+            self.presentationMode.wrappedValue.dismiss()
+        }) {
+            Image(systemName: "xmark.circle.fill")
+                .style(appStyle: .barButton)
+                .foregroundColor(.red)
+        }
+        .buttonStyle(BorderedBarButtonStyle())
+        .accentColor(Color.red.opacity(0.2))
+    }
+
+    private var saveButton: some View {
+        Button(action: {
+            if self.viewModel.design.title.isEmpty {
+                self.nameErrorBorderColor = .red
+            }
+
+            if !self.viewModel.design.hasValidCode {
+                self.codeErrorBorderColor = .red
+            }
+
+            guard
+                !self.viewModel.design.title.isEmpty,
+                self.viewModel.design.hasValidCode else {
+                    return
+            }
+
+            self.viewModel.save()
+            self.presentationMode.wrappedValue.dismiss()
+        }) {
+            Image(systemName: "checkmark.seal.fill")
+                .style(appStyle: .barButton)
+                .foregroundColor(.acTabBarBackground)
+        }
+        .buttonStyle(BorderedBarButtonStyle())
+        .accentColor(Color.acTabBarBackground.opacity(0.2))
+    }
+}
+
+// MARK: - Preview
+
+#if DEBUG
+struct DesignFormView_Previews: PreviewProvider {
+    static var previews: some View {
+        Group {
+            DesignFormView(viewModel: DesignFormViewModel(design: nil))
+
+            DesignFormView(viewModel: DesignFormViewModel(design: Design(title: "Jedi",
+                                                                         code: "MOPJ15LTDSXC4T",
+                                                                         description: "Jedi Tunic")))
+                .preferredColorScheme(.dark)
+        }
+    }
+}
+#endif
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignListView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignListView.swift
new file mode 100644
index 00000000..745372cc
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignListView.swift
@@ -0,0 +1,52 @@
+//
+//  DesignListView.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 27.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import SwiftUI
+import Backend
+
+struct DesignListView: View {
+
+    // MARK: - Properties
+
+    @EnvironmentObject private var collection: UserCollection
+    @State private var sheet: Sheet.SheetType?
+
+    // MARK: - Public
+
+    var body: some View {
+        List {
+            Button(action: {
+                self.sheet = .designForm(editingDesign: nil)
+            }, label: {
+                Text("Add Creator/Design Item")
+                    .foregroundColor(.acHeaderBackground)
+            })
+
+            ForEach(collection.designs) { design in
+                DesignRowView(viewModel: DesignRowViewModel(design: design))
+            }.onDelete { indexes in
+                self.collection.deleteDesign(at: indexes.first!)
+            }
+        }
+        .navigationBarTitle(Text("Designs"), displayMode: .automatic)
+        .sheet(item: $sheet, content: { Sheet(sheetType: $0) })
+    }
+}
+
+// MARK: - Preview
+
+#if DEBUG
+struct DesignListView_Previews: PreviewProvider {
+    static var previews: some View {
+        NavigationView {
+            DesignListView()
+        }
+        .environmentObject(UserCollection.shared)
+    }
+}
+#endif
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignRowView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignRowView.swift
new file mode 100644
index 00000000..cec961e3
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/design/DesignRowView.swift
@@ -0,0 +1,73 @@
+//
+//  DesignListView.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 24.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import SwiftUI
+import Backend
+
+struct DesignRowView: View {
+
+    // MARK: - Properties
+
+    let viewModel: DesignRowViewModel
+
+    // MARK: - Life cycle
+
+    init(viewModel: DesignRowViewModel) {
+        self.viewModel = viewModel
+    }
+
+    // MARK: - Public
+
+    var body: some View {
+        HStack {
+            VStack(alignment: .leading, spacing: 2) {
+                Text(viewModel.title.capitalized)
+                    .style(appStyle: .rowTitle)
+
+                Text(viewModel.code)
+                    .fontWeight(.semibold)
+                    .font(.subheadline)
+
+                if !viewModel.description.isEmpty {
+                    Text(viewModel.description.capitalized)
+                        .style(appStyle: .rowDescription)
+                }
+            }
+
+            Spacer()
+
+            Text(viewModel.category)
+                .font(.caption)
+                .fontWeight(.bold)
+                .foregroundColor(.acHeaderBackground)
+                .lineLimit(1)
+        }
+        .foregroundColor(.acSecondaryText)
+    }
+}
+
+// MARK: - Preview
+
+#if DEBUG
+struct DesignRowView_Previews: PreviewProvider {
+    static var previews: some View {
+        NavigationView {
+            List {
+                DesignRowView(viewModel:
+                    DesignRowViewModel(design: Design(title: "Jedi", code: "MOPJ15LTDSXC4T", description: "Jedi Tunic")))
+
+                DesignRowView(viewModel:
+                    DesignRowViewModel(design: Design(title: "Sam",
+                                                      code: "MA667931515180",
+                                                      description: "Sokola Island. Might have more StarWars designs"))
+                )
+            }
+        }
+    }
+}
+#endif
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/message/MessageView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/message/MessageView.swift
new file mode 100644
index 00000000..e0f5c363
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/message/MessageView.swift
@@ -0,0 +1,57 @@
+//
+//  EmptyView.swift
+//  ACHNBrowserUI
+//
+//  Created by Otavio Cordeiro on 27.05.20.
+//  Copyright © 2020 Thomas Ricouard. All rights reserved.
+//
+
+import SwiftUI
+
+struct MessageView: View {
+
+    // MARK: - Properties
+
+    private let string: String
+
+    // MARK: - Life cycle
+
+    init(string: String) {
+        self.string = string
+    }
+
+    init(collectionName: String) {
+        self.string = String.init(format: NSLocalizedString("When you stars some %@, they'll be displayed here.",
+                                                            comment: ""), collectionName)
+    }
+
+    init(noResultsFor string: String) {
+        self.string = String.init(format: NSLocalizedString("No results for %@", comment: ""), string)
+    }
+
+    // MARK: - Public
+
+    var body: some View {
+        Text(string).foregroundColor(.acSecondaryText)
+    }
+}
+
+// MARK: - Preview
+
+#if DEBUG
+struct CollectionEmptyView_Previews: PreviewProvider {
+    static var previews: some View {
+        Group {
+            MessageView(string: "Lorem Ipsum")
+                .previewLayout(.sizeThatFits)
+
+            MessageView(collectionName: "Pan-dimensional Mice")
+                .previewLayout(.sizeThatFits)
+
+            MessageView(collectionName: "Vogons")
+                .previewLayout(.sizeThatFits)
+                .preferredColorScheme(.dark)
+        }
+    }
+}
+#endif
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/shared/Sheet.swift b/ACHNBrowserUI/ACHNBrowserUI/views/shared/Sheet.swift
index 9d365330..acfdfd94 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/views/shared/Sheet.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/shared/Sheet.swift
@@ -18,7 +18,8 @@ struct Sheet: View {
         case turnipsForm(subManager: SubscriptionManager)
         case subscription(source: SubscribeView.Source, subManager: SubscriptionManager)
         case settings(subManager: SubscriptionManager, collection: UserCollection)
-        
+        case designForm(editingDesign: Design?)
+
         var id: String {
             switch self {
             case .safari(let url):
@@ -37,6 +38,8 @@ struct Sheet: View {
                 return "sub"
             case .userListForm:
                 return "userListForm"
+            case .designForm:
+                return "designForm"
             }
         }
     }
@@ -64,6 +67,9 @@ struct Sheet: View {
             return AnyView(SubscribeView(source: source).environmentObject(subManager))
         case .userListForm(let list):
             return AnyView(UserListFormView(editingList: list))
+        case .designForm(let design):
+            let viewModel = DesignFormViewModel(design: design)
+            return AnyView(DesignFormView(viewModel: viewModel))
         case .rearrange(let viewModel):
             return AnyView(NavigationView {
                 TodaySectionEditView(viewModel: viewModel)
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/userLists/UserListDetailView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/userLists/UserListDetailView.swift
index 65ff0cfe..bcd08b20 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/views/userLists/UserListDetailView.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/userLists/UserListDetailView.swift
@@ -73,15 +73,13 @@ struct UserListDetailView: View {
                             self.viewModel.deleteItem(at: indexes.first!)
                         }
                     } else {
-                        Text("Items added to your list from the search will be displayed there.")
-                            .foregroundColor(.acSecondaryText)
+                        MessageView(string: "Items added to your list from the search will be displayed there.")
                     }
                 } else {
                     if searchViewModdel.isLoadingData {
                         RowLoadingView(isLoading: $isLoadingData).animation(.default)
                     } else if searchCategories.isEmpty {
-                        Text("No results for \(searchViewModdel.searchText)")
-                            .foregroundColor(.acSecondaryText)
+                        MessageView(noResultsFor: searchViewModdel.searchText)
                     } else {
                         ForEach(searchCategories, id: \.0, content: searchSection)
                     }
