diff --git a/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj b/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
index b6eac8b2..a2556c41 100644
--- a/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
+++ b/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
@@ -22,6 +22,7 @@
 		4C30BFC3249627FD003E96A2 /* Sequence+Unique.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4C30BFC2249627FC003E96A2 /* Sequence+Unique.swift */; };
 		4C6E95FD24842F690074433B /* Collection+SafeDirectAccess.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4C6E95FC24842F690074433B /* Collection+SafeDirectAccess.swift */; };
 		4C7F555D248B91C80089F26C /* TodayVillagerVisitsSection.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4C7F555C248B91C80089F26C /* TodayVillagerVisitsSection.swift */; };
+		4CB1E3D925A0BB8F0021A175 /* VariantsForLikeView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4CB1E3D825A0BB8F0021A175 /* VariantsForLikeView.swift */; };
 		69157B7E2471A5A1005B9002 /* TodayMysteryIslandsSection.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69157B6D247121ED005B9002 /* TodayMysteryIslandsSection.swift */; };
 		69157B802471A5A1005B9002 /* UserListFormView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 69875F5424654EB900B1D46C /* UserListFormView.swift */; };
 		69157B812471A5A1005B9002 /* RowLoadingView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 693E2BBD2465B65D00B85CB8 /* RowLoadingView.swift */; };
@@ -252,6 +253,7 @@
 		4C7F2F772461F10300930928 /* TurnipsChartVerticalLegend.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = TurnipsChartVerticalLegend.swift; sourceTree = "<group>"; };
 		4C7F555C248B91C80089F26C /* TodayVillagerVisitsSection.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = TodayVillagerVisitsSection.swift; sourceTree = "<group>"; };
 		4C7FE788245B57A10011E8AB /* AdaptsToSoftwareKeyboard.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = AdaptsToSoftwareKeyboard.swift; sourceTree = "<group>"; };
+		4CB1E3D825A0BB8F0021A175 /* VariantsForLikeView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = VariantsForLikeView.swift; sourceTree = "<group>"; };
 		4CF14B74246B3E9E00F740BF /* TurnipsChartValuesView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = TurnipsChartValuesView.swift; sourceTree = "<group>"; };
 		4CF8049A24635E54005C6C63 /* Path.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = Path.swift; sourceTree = "<group>"; };
 		4F92A4522470AC01003A2911 /* CollectionRowView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CollectionRowView.swift; sourceTree = "<group>"; };
@@ -851,6 +853,7 @@
 				69CA5F00244F1AD300B275B6 /* SectionHeaderView.swift */,
 				69875F4F24652A4700B1D46C /* Sheet.swift */,
 				69E67EFA2494C35B00ED7AC8 /* ButtonImageCounterOverlay.swift */,
+				4CB1E3D825A0BB8F0021A175 /* VariantsForLikeView.swift */,
 			);
 			path = shared;
 			sourceTree = "<group>";
@@ -1233,6 +1236,7 @@
 				69157BB02471A5A1005B9002 /* TodayView.swift in Sources */,
 				69157BB12471A5A1005B9002 /* CategoriesView.swift in Sources */,
 				69157BB22471A5A1005B9002 /* Sheet.swift in Sources */,
+				4CB1E3D925A0BB8F0021A175 /* VariantsForLikeView.swift in Sources */,
 				6950EB37249D03C7002A4902 /* CommentsSection.swift in Sources */,
 				69157BB32471A5A1005B9002 /* CategoriesSearchViewModel.swift in Sources */,
 				69157BB42471A5A1005B9002 /* UserListRow.swift in Sources */,
diff --git a/ACHNBrowserUI/ACHNBrowserUI/viewModels/LikeButtonViewModel.swift b/ACHNBrowserUI/ACHNBrowserUI/viewModels/LikeButtonViewModel.swift
index 80d9cd2a..88c62219 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/viewModels/LikeButtonViewModel.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/viewModels/LikeButtonViewModel.swift
@@ -18,6 +18,7 @@ class LikeButtonViewModel: ObservableObject {
     let villager: Villager?
     
     @Published var isInCollection = false
+    @Published var variantsCompletionStatus: VariantsCompletionStatus = .unstarted
     
     private let collection: UserCollection
     private var cancellable: AnyCancellable?
@@ -42,6 +43,11 @@ class LikeButtonViewModel: ObservableObject {
                         self.isInCollection = variants.contains(for: item, variant: variant)
                         return
                     }
+                    if item.hasSomeVariations {
+                        self.variantsCompletionStatus = variants.completionStatus(for: item)
+                        self.isInCollection = self.variantsCompletionStatus != .unstarted
+                        return
+                    }
                     self.isInCollection = items.contains(item) || critters.contains(item)
                     return
                 } else if let villager = villager {
@@ -70,4 +76,8 @@ class LikeButtonViewModel: ObservableObject {
         }
         return added
     }
+
+    var hasSomeVariations: Bool {
+        variant == nil && item?.hasSomeVariations == true
+    }
 }
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/items/ItemsView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/items/ItemsView.swift
index 56c5b3d9..c2c3192f 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/views/items/ItemsView.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/items/ItemsView.swift
@@ -29,6 +29,7 @@ struct ItemsView: View {
     
     @StateObject var viewModel: ItemsViewModel
     @State private var contentMode: ContentMode = .listLarge
+    @State private var likedItemWithVariants: Item?
     let customTitle: String?
     
     init(category: Backend.Category, items: [Item]? = nil, keyword: String? = nil) {
@@ -110,30 +111,36 @@ struct ItemsView: View {
     
     @ViewBuilder
     private var contentView: some View {
-        if contentMode == .grid {
-            ScrollView {
-                LazyVGrid(columns: [GridItem(.adaptive(minimum: 120), spacing: 16)], spacing: 16) {
-                    ForEach(viewModel.items) { item in
-                        ItemGridItemView(item: item)
+        ZStack {
+            if contentMode == .grid {
+                ScrollView {
+                    LazyVGrid(columns: [GridItem(.adaptive(minimum: 120), spacing: 16)], spacing: 16) {
+                        ForEach(viewModel.items) { item in
+                            ItemGridItemView(item: item)
+                        }
                     }
-                }
-                .padding(8)
-                .background(Color.acBackground)
-            }.background(Color.acBackground)
-        } else {
-            List {
-                Section(header: SearchField(searchText: $viewModel.searchText)) {
-                    ForEach(viewModel.items) { item in
-                        NavigationLink(destination: LazyView(ItemDetailView(item: item))) {
-                            ItemRowView(displayMode: contentMode == .listLarge ? .large : .compact,
-                                        item: item)
+                    .padding(8)
+                    .background(Color.acBackground)
+                }.background(Color.acBackground)
+            } else {
+                List {
+                    Section(header: SearchField(searchText: $viewModel.searchText)) {
+                        ForEach(viewModel.items) { item in
+                            NavigationLink(destination: LazyView(ItemDetailView(item: item))) {
+                                ItemRowView(
+                                    displayMode: contentMode == .listLarge ? .large : .compact,
+                                    item: item,
+                                    likedItemWithVariants: $likedItemWithVariants
+                                )
+                            }
+                            .listRowBackground(Color.acSecondaryBackground)
                         }
-                        .listRowBackground(Color.acSecondaryBackground)
                     }
                 }
+                .listStyle(GroupedListStyle())
+                .id(viewModel.sort)
             }
-            .listStyle(GroupedListStyle())
-            .id(viewModel.sort)
+            VariantsForLikeView(item: $likedItemWithVariants)
         }
     }
 }
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/items/detail/ItemDetailView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/items/detail/ItemDetailView.swift
index 1c3ad72f..e1ca237a 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/views/items/detail/ItemDetailView.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/items/detail/ItemDetailView.swift
@@ -20,6 +20,7 @@ struct ItemDetailView: View {
     
     @State private var displayedVariant: Variant?
     @State private var selectedSheet: Sheet.SheetType?
+    @State private var itemToDisplayVariantsLike: Item?
 
     init(item: Item) {
         self._itemViewModel = StateObject(wrappedValue: ItemDetailViewModel(item: item))
@@ -43,9 +44,16 @@ struct ItemDetailView: View {
         
         return [ItemDetailSource(name: itemViewModel.item.localizedName.capitalized, image: image)]
     }
-    
+
     // MARK: - Boby
     var body: some View {
+        ZStack {
+            content
+            VariantsForLikeView(item: $itemToDisplayVariantsLike)
+        }
+    }
+
+    var content: some View {
         List {
             ItemDetailInfoView(item: itemViewModel.item,
                                recipe: itemViewModel.recipe,
@@ -123,12 +131,22 @@ extension ItemDetailView {
     
     private var navButtons: some View {
         HStack {
-            LikeButtonView(item: itemViewModel.item,
-                           variant: displayedVariant).imageScale(.large)
-                .safeHoverEffectBarItem(position: .trailing)
+            LikeButtonView(
+                item: itemViewModel.item,
+                variant: displayedVariant,
+                customAction: tapOnMainLikeButton
+            )
+            .imageScale(.large)
+            .safeHoverEffectBarItem(position: .trailing)
+
             shareButton
         }
     }
+
+    private func tapOnMainLikeButton() {
+        // Display variants bottom sheet only if the item has some variants
+        itemToDisplayVariantsLike = itemViewModel.item.hasSomeVariations ? itemViewModel.item : nil
+    }
     
     private var keywordsSection: some View {
         Section(header: SectionHeaderView(text: "Keywords", icon: "link")) {
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/items/row/ItemRowView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/items/row/ItemRowView.swift
index bfc8032a..ec7be3e6 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/views/items/row/ItemRowView.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/items/row/ItemRowView.swift
@@ -17,8 +17,18 @@ struct ItemRowView: View {
     
     let displayMode: DisplayMode
     let item: Item
-    
+    @Binding var likedItemWithVariants: Item?
     @State private var displayedVariant: Variant?
+
+    init(
+        displayMode: DisplayMode,
+        item: Item,
+        likedItemWithVariants: Binding<Item?> = .constant(nil)
+    ) {
+        self.displayMode = displayMode
+        self.item = item
+        self._likedItemWithVariants = likedItemWithVariants
+    }
     
     private var imageSize: CGFloat {
         switch displayMode {
@@ -87,7 +97,11 @@ struct ItemRowView: View {
     var body: some View {
         HStack(spacing: 8) {
             if displayMode != .largeNoButton {
-                LikeButtonView(item: item, variant: displayedVariant)
+                LikeButtonView(
+                    item: item,
+                    variant: displayedVariant,
+                    likedItemWithVariants: $likedItemWithVariants
+                )
             }
             if item.finalImage == nil && displayedVariant == nil {
                 Image(item.appCategory.iconName())
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/shared/LikeButtonView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/shared/LikeButtonView.swift
index c0bc1d71..68154a7a 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/views/shared/LikeButtonView.swift
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/shared/LikeButtonView.swift
@@ -11,13 +11,23 @@ import Backend
 
 struct LikeButtonView: View {
     @StateObject private var viewModel: LikeButtonViewModel
+    @Binding private var likedItemWithVariants: Item?
+    private var customAction: () -> Void = { }
     
-    init(item: Item, variant: Variant?) {
+    init(
+        item: Item,
+        variant: Variant?,
+        likedItemWithVariants: Binding<Item?> = .constant(nil),
+        customAction: @escaping () -> Void = { }
+    ) {
         _viewModel = StateObject(wrappedValue: LikeButtonViewModel(item: item, variant: variant))
+        _likedItemWithVariants = likedItemWithVariants
+        self.customAction = customAction
     }
     
     init(villager: Villager) {
         _viewModel = StateObject(wrappedValue: LikeButtonViewModel(villager: villager))
+        _likedItemWithVariants = .constant(nil)
     }
         
     var imageName: String {
@@ -25,6 +35,13 @@ struct LikeButtonView: View {
             if viewModel.item?.isCritter == true {
                 return viewModel.isInCollection ? "checkmark.seal.fill" : "checkmark.seal"
             } else {
+                if viewModel.hasSomeVariations {
+                    switch viewModel.variantsCompletionStatus {
+                    case .unstarted: return "star"
+                    case .partial: return "star.leadinghalf.fill"
+                    case .complete: return "star.fill"
+                    }
+                }
                 return viewModel.isInCollection ? "star.fill" : "star"
             }
         } else {
@@ -44,8 +61,13 @@ struct LikeButtonView: View {
     
     var body: some View {
         Button(action: {
-            let added = self.viewModel.toggleCollection()
-            FeedbackGenerator.shared.triggerNotification(type: added ? .success : .warning)
+            if viewModel.hasSomeVariations {
+                likedItemWithVariants = viewModel.item
+            } else {
+                let added = self.viewModel.toggleCollection()
+                FeedbackGenerator.shared.triggerNotification(type: added ? .success : .warning)
+            }
+            customAction()
         }) {
             Image(systemName: imageName).foregroundColor(color)
         }
diff --git a/ACHNBrowserUI/ACHNBrowserUI/views/shared/VariantsForLikeView.swift b/ACHNBrowserUI/ACHNBrowserUI/views/shared/VariantsForLikeView.swift
new file mode 100644
index 00000000..88a82b7f
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/views/shared/VariantsForLikeView.swift
@@ -0,0 +1,67 @@
+//
+//  VariantsForLikeView.swift
+//  ACHNBrowserUI
+//
+//  Created by Renaud JENNY on 02/01/2021.
+//  Copyright © 2021 Thomas Ricouard. All rights reserved.
+//
+
+import SwiftUI
+import Backend
+import UI
+
+struct VariantsForLikeView: View {
+    @Binding var item: Item?
+
+    var body: some View {
+        ZStack {
+            Color.black
+                .opacity(item != nil ? 1/6 : 0)
+                .onTapGesture { item = nil }
+            if let item = item {
+                VStack {
+                    Spacer()
+                    VStack {
+                        SectionHeaderView(text: "Variants", icon: "paintbrush.fill").padding()
+                        ScrollView(.horizontal, showsIndicators: false) {
+                            HStack {
+                                ForEach(item.variations ?? []) { variant in
+                                    ZStack(alignment: .topLeading) {
+                                        ItemImage(path: variant.content.image, size: 75)
+                                        LikeButtonView(item: item, variant: variant)
+                                    }
+                                }
+                            }.padding()
+                        }
+                        .listRowInsets(EdgeInsets())
+                    }
+                    .background(Color.acSecondaryBackground.shadow(radius: 2, x: 0, y: -2))
+                }
+                .transition(.move(edge: .bottom))
+            }
+        }
+        .animation(.easeInOut)
+    }
+}
+
+#if DEBUG
+struct VariantsForLikeView_Previews: PreviewProvider {
+    static var previews: some View {
+        Preview()
+    }
+
+    private struct Preview: View {
+        @State private var item: Item?
+
+        var body: some View {
+            ZStack {
+                Button { item = static_item } label: {
+                    Text("Show Variants")
+                }
+                VariantsForLikeView(item: $item)
+            }
+        }
+    }
+}
+
+#endif
diff --git a/ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift b/ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift
index f5642f19..f27f3fc6 100644
--- a/ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift
+++ b/ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift
@@ -136,18 +136,29 @@ public class UserCollection: ObservableObject {
         guard let filename = item.filename else {
             return false
         }
+        let isAdded: Bool
         if variants[filename]?.contains(variant) == true {
             variants[filename]?.removeAll(where: { $0 == variant })
             save()
-            return false
+            isAdded = false
         } else {
             if variants[filename] == nil {
                 variants[filename] = []
             }
             variants[filename]?.append(variant)
             save()
-            return true
+            isAdded = true
+        }
+
+        if !items.contains(item) && variants.completionStatus(for: item) != .unstarted {
+            // We add the item to the liked items if at least one of its variant is added
+            _ = toggleItem(item: item)
+        } else if items.contains(item) && variants.completionStatus(for: item) == .unstarted {
+            // We remove the item from the liked items if there is none of its variants liked
+            _ = toggleItem(item: item)
         }
+
+        return isAdded
     }
     
     public func toggleCritters(critter: Item) -> Bool {
@@ -499,10 +510,37 @@ public class UserCollection: ObservableObject {
 }
 
 public extension Dictionary where Key == String, Value == [Variant] {
-    public func contains(for item: Item, variant: Variant) -> Bool {
+    func contains(for item: Item, variant: Variant) -> Bool {
         guard let filename = item.filename else {
             return false
         }
         return self[filename]?.contains(variant) == true
     }
+
+    func completionStatus(for item: Item) -> VariantsCompletionStatus {
+        guard let filename = item.filename,
+              let itemVariations = item.variations,
+              let currentVariations = self[filename]
+        else { return .unstarted }
+
+        if currentVariations.count <= 0 {
+            return .unstarted
+        } else if currentVariations.count < itemVariations.count {
+            return .partial
+        } else {
+            return .complete
+        }
+    }
+}
+
+public enum VariantsCompletionStatus {
+    case unstarted
+    case partial
+    case complete
+}
+
+public extension Item {
+    var hasSomeVariations: Bool {
+        variations?.count ?? 0 > 1
+    }
 }
diff --git a/ACHNBrowserUI/Packages/Backend/Sources/Backend/Models/Item.swift b/ACHNBrowserUI/Packages/Backend/Sources/Backend/Models/Item.swift
index 46b91a9e..0a67cd91 100644
--- a/ACHNBrowserUI/Packages/Backend/Sources/Backend/Models/Item.swift
+++ b/ACHNBrowserUI/Packages/Backend/Sources/Backend/Models/Item.swift
@@ -272,7 +272,20 @@ public let static_item = Item(name: "Acoustic guitar",
                        dIY: true,
                        customize: true,
                        movementSpeed: nil,
-                       variations: nil,
+                       variations: [
+                            Variant(id: 49134, content: Variant.Content(
+                                image: "https://acnhcdn.com/latest/FtrIcon/FtrAcorsticguitar_Remake_0_0.png",
+                                colors: ["Beige", "Brown"]
+                            )),
+                            Variant(id: 49135, content: Variant.Content(
+                                image: "https://acnhcdn.com/latest/FtrIcon/FtrAcorsticguitar_Remake_1_0.png",
+                                colors: ["Red", "Black"]
+                            )),
+                            Variant(id: 49136, content: Variant.Content(
+                                image: "https://acnhcdn.com/latest/FtrIcon/FtrAcorsticguitar_Remake_2_0.png",
+                                colors: ["Brown", "Black"]
+                            )),
+                       ],
                        category: "Housewares",
                        materials: nil,
                        buy: 200,
